misc-utils/uuidd.c

```diff
@@ -156,15 +161,15 @@ static int call_daemon(const char *socket_path, int op, char *buf,
 	}
 
 	if (op == UUIDD_OP_BULK_RANDOM_UUID) {
-		if ((*num) * UUID_LEN > buflen - 4)
-			*num = (buflen - 4) / UUID_LEN;
+		if ((buflen - UUID_NUM_LEN) < (size_t)((*num) * UUID_DATA_LEN))
+			*num = (buflen - UUID_NUM_LEN) / UUID_DATA_LEN;
 	}
 	op_buf[0] = op;
-	op_len = 1;
+	op_len = UUID_OP_LEN;
 	if ((op == UUIDD_OP_BULK_TIME_UUID) ||
 	    (op == UUIDD_OP_BULK_RANDOM_UUID)) {
-		memcpy(op_buf + 1, num, sizeof(int));
-		op_len += sizeof(int);
+		memcpy(op_buf + UUID_OP_LEN, num, UUID_NUM_LEN);
+		op_len += UUID_NUM_LEN;
 	}
```

Comment:
IMHO it's always better and robust to follow variables when use sizeof() for memcpy().
```
memcpy(op_buf + UUID_OP_LEN, num, sizeof(num));
op_len += sizeof(num);
```